src/poetry/vcs/git/backend.py

```diff
@@ -319,23 +320,35 @@ def _clone(cls, url: str, refspec: GitRefSpec, target: Path) -> Repo:
         return local
 
     @classmethod
-    def _clone_submodules(cls, repo: Repo) -> None:
+    def _clone_submodules(cls, repo: Repo, main_url: str) -> None:
         """
         Helper method to identify configured submodules and clone them recursively.
         """
         repo_root = Path(repo.path)
         modules_config = repo_root.joinpath(".gitmodules")
 
+        # A relative URL by definition starts with ../ or ./
+        relative_submodule_regex = re.compile("^(\\.\\./|\\./),?")
```

Comment:
This seems unnecessary, since urljoin will already ignore the first argument if the second argument is already an absolute URL:

```
>>> from urllib.parse import urljoin
>>> urljoin('https://example.com/some/base/', '../bar')
'https://example.com/some/bar'
>>> urljoin('https://example.com/some/base/', 'https://voorbeeld.info/blah/')
'https://voorbeeld.info/blah/'
```